Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfigurationValidator.cs

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using EdFi.Common.Configuration;
+
+namespace EdFi.LoadTools.BulkLoadClient.Application
+{
+    public class BulkLoadClientConfigurationValidator : IBulkLoadClientConfigurationValidator
+    {
+        private readonly BulkLoadClientConfiguration _configuration;
+
+        public BulkLoadClientConfigurationValidator(BulkLoadClientConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public string ErrorText { get; private set; }
+
+        public bool IsValid()
+        {
+            bool result =
+                !(string.IsNullOrEmpty(_configuration.ApiUrl)
+                  || string.IsNullOrEmpty(_configuration.WorkingFolder)
+                  || string.IsNullOrEmpty(_configuration.DataFolder)
+                  || string.IsNullOrEmpty(_configuration.XsdFolder)
+                  || string.IsNullOrEmpty(_configuration.MetadataUrl)
+                  || string.IsNullOrEmpty(_configuration.OAuthKey)
+                  || string.IsNullOrEmpty(_configuration.OAuthSecret)
+                  || string.IsNullOrEmpty(_configuration.OauthUrl))
+                && Directory.Exists(_configuration.DataFolder)
+                && Uri.IsWellFormedUriString(_configuration.ApiUrl, UriKind.Absolute)
+                && Uri.IsWellFormedUriString(_configuration.MetadataUrl, UriKind.Absolute)
+                && Uri.IsWellFormedUriString(_configuration.OauthUrl, UriKind.Absolute);
+
+
+            if (_configuration.ApiMode == ApiMode.YearSpecific)
+            {
+                result = result && _configuration.SchoolYear.HasValue;
+            }
+
+            if (!result)
+            {
+                ErrorText = CreateErrorText();
+            }
+
+            return result;
+        }
+
+        private string CreateErrorText()
+        {
+            var sb = new StringBuilder();
+
+            if (string.IsNullOrEmpty(_configuration.OAuthKey))
+            {
+                sb.AppendLine("Option 'k:key' parse error. missing value.");
+            }
+
+            if (string.IsNullOrEmpty(_configuration.OAuthSecret))
+            {
+                sb.AppendLine("Option 's:secret' parse error. missing value.");
+            }
+
+            if (string.IsNullOrEmpty(_configuration.ApiUrl) || !Uri.IsWellFormedUriString(_configuration.ApiUrl, UriKind.Absolute))
+            {
+                sb.AppendLine($"Option 'a:apiurl' parse error. \"{_configuration.ApiUrl}\" is not a url.");
```

Comment:
Add single quotes around parameter values in messages.

Reply:
done